src/cacheEngine.ts

```diff
@@ -1,23 +1,30 @@
-import startOfMonth from 'date-fns/startOfMonth'
-import sub from 'date-fns/sub'
 import nano from 'nano'
 
 import { getAnalytics } from './apiAnalytics'
 import { config } from './config'
 import { asDbReq } from './dbutils'
 import { initDbs } from './initDbs'
 import { asApps } from './types'
-import { datelog, snooze } from './util'
+import {
+  datelog,
+  getStartOfMonthsAgo,
+  getStartOfMonthsFromNow,
+  snooze
+} from './util'
 
 const CACHE_UPDATE_LOOKBACK_MONTHS = config.cacheLookbackMonths ?? 3
 
 const BULK_WRITE_SIZE = 50
 const UPDATE_FREQUENCY_MS = 1000 * 60 * 30
-const CACHE_UPDATE_BLOCK_S = 60 * 60 * 24 * 60 // 60 days
+const CACHE_UPDATE_BLOCK_MONTHS = 3
 
 const nanoDb = nano(config.couchDbFullpath)
 
 const TIME_PERIODS = ['hour', 'day', 'month']
+const birthdayStart = getStartOfMonthsAgo(
+  new Date('2017-01Z').toISOString(),
+  0
+).toISOString()
```

Comment:
I think we're using this 2017 date commonly as a epoch birthday of some kind that we start querying from or back to... Maybe it would make sense to move these constants to a file of all the various types of these dates (string, ms timestamp, second timestamps, date object, etc), and then we can comment on the reasoning behind this 2017 date. That'd be helpful for documentation purpose and re-use.

Reply:
Sort of, but I do change the date for different plugins since I don't got as far back as 2017 if I for sure know that we added the provider in 2022.